lib/src/source/git.dart

```diff
@@ -436,15 +436,25 @@ class BoundGitSource extends CachedSource {
   }
 
   ///  Clean-up [dirPath] if it's an invalid git repository
-  void _cleanInvalidGitRepoCache(String dirPath) {
-    if (dirExists(dirPath)) {
-      var processResult = runProcessSync(
-          git.command, ['rev-parse', '--is-inside-git-dir'],
-          workingDir: dirPath);
-      var result = processResult.stdout?.join('\n');
-      if (processResult.exitCode != 0 || result != 'true') {
-        deleteEntry(dirPath);
+  Future<void> _cleanInvalidGitRepoCache(String dirPath) async {
+    if (!dirExists(dirPath)) {
+      return;
+    }
+    var isValid = true;
```

Comment:
Is this ever false?

Reply:
Do we have tests of this?

Reply:
Yes rarely, and yes,

But we have no tests for what happens when if you do this when `git` is not installed.

Reply:
I guess one could make the argument, that we should try desired git operations first, and only if that fails, validate if the directory is a valid git clone... and if not retry after deleting.

Reply:
> But we have no tests for what happens when if you do this when git is not installed.

Is it possible to add such a test to avoid regressing on this fix?